addons/mail/static/src/models/thread_view/thread_view.js

```diff
@@ -113,11 +113,37 @@ function factory(dependencies) {
             }
         }
 
+        /**
+         * Fetches the messages to be displayed by `this`, if applicable.
+         * This method should be called to ensure the latest messages are
+         * fetched when displaying an existing non-main thread cache.
+         * This is necessary because only the main cache is automatically
+         * updated with server changes. Other caches only contain the messages
+         * at the point in time when their last fetch was called.
+         */
+        fetchNonMainCacheMessages() {
+            if (!this.threadCache) {
+                return;
+            }
+            if (this.thread.isTemporary) {
+                // Temporary thread has no server messages.
+                return;
+            }
+            if (this.thread.mainCache === this.threadCache) {
+                // The main cache is automatically sync with server updates.
+                return;
+            }
+            if (!this.threadCache.isLoading) {
+                this.threadCache.update({ hasToLoadMessages: true });
+            }
+        }
+
```

Comment:
Doesn't look like a "fetch" method. Makes its use completely cryptic for me...
Please modify use of this function, either with just a rename or better code splitting.
Will look at places where this function is called after this change.

Reply:
`hasToLoadMessages` is the way to instruct thread cache to do a fetch (load messages).

`instructNonMainCacheToReloadMessages?`

Reply:
I don't think just method rename is enough. Calling this method has many inappropriate use of pre-emptive return. 

Good usage of preemptive return are returning something meaningful of the function. These are just plain `return;`, which leads to bad calling of this method in the first place. This is a sign of bad code breakdown.

Reply:
Made new code split, everything now handled in thread cache as it should (and was already partially done).